src/lsdb/catalog/catalog.py

```diff
@@ -68,3 +80,91 @@ def get_partition_index(self, order: int, pixel: int) -> int:
             raise ValueError(f"Pixel at order {order} pixel {pixel} not in Catalog")
         partition_index = self._ddf_pixel_map[hp_pixel]
         return partition_index
+
+    @property
+    def name(self):
+        """The name of the catalog"""
+        return self.hc_structure.catalog_name
+
+    def crossmatch(
+        self,
+        other: Catalog,
+        suffixes: Tuple[str, str] | None = None,
+        algorithm: Type[AbstractCrossmatchAlgorithm]
+        | BuiltInCrossmatchAlgorithm = BuiltInCrossmatchAlgorithm.KD_TREE,
+        name: str | None = None,
```

Comment:
```suggestion
        output_catalog_name: str | None = None,
```

Give a more descriptive name.